src/nvim/os/env.c

```diff
@@ -87,3 +98,591 @@ void os_get_hostname(char *hostname, size_t len)
+          free(var);
+        }
+        var = p;
+        mustfree = true;
+        forward_slash(var);
+      }
+#endif
+
+      /* If "var" contains white space, escape it with a backslash.
+       * Required for ":e ~/tt" when $HOME includes a space. */
+      if (esc && var != NULL && vim_strpbrk(var, (char_u *)" \t") != NULL) {
+        char_u  *p = vim_strsave_escaped(var, (char_u *)" \t");
+
+        if (mustfree)
+          free(var);
+        var = p;
+        mustfree = true;
+      }
+
+      if (var != NULL && *var != NUL
+          && (STRLEN(var) + STRLEN(tail) + 1 < (unsigned)dstlen)) {
+        STRCPY(dst, var);
+        dstlen -= (int)STRLEN(var);
+        c = (int)STRLEN(var);
+        /* if var[] ends in a path separator and tail[] starts
+         * with it, skip a character */
+        if (*var != NUL && after_pathsep(dst, dst + c)
+#if defined(BACKSLASH_IN_FILENAME)
+            && dst[-1] != ':'
+#endif
+            && vim_ispathsep(*tail))
+          ++tail;
+        dst += c;
+        src = tail;
+        copy_char = false;
+      }
+      if (mustfree)
+        free(var);
+    }
+
+    if (copy_char) {        /* copy at least one char */
+      /*
+       * Recognize the start of a new name, for '~'.
+       * Don't do this when "one" is true, to avoid expanding "~" in
+       * ":edit foo ~ foo".
+       */
+      at_start = false;
+      if (src[0] == '\\' && src[1] != NUL) {
+        *dst++ = *src++;
+        --dstlen;
+      } else if ((src[0] == ' ' || src[0] == ',') && !one) {
+        at_start = true;
+      }
+      *dst++ = *src++;
+      --dstlen;
+
+      if (startstr != NULL && src - startstr_len >= srcp
+          && STRNCMP(src - startstr_len, startstr, startstr_len) == 0)
+        at_start = true;
+    }
+  }
+  *dst = NUL;
+}
+
+/// Check if the directory "vimdir/<version>" or "vimdir/runtime" exists.
+/// Return NULL if not, return its name in allocated memory otherwise.
+/// @param vimdir directory to test
+static char_u *vim_version_dir(char_u *vimdir)
+{
+  char_u      *p;
+
+  if (vimdir == NULL || *vimdir == NUL)
+    return NULL;
+  p = concat_fnames(vimdir, (char_u *)VIM_VERSION_NODOT, true);
+  if (os_isdir(p))
+    return p;
+  free(p);
+  p = concat_fnames(vimdir, (char_u *)RUNTIME_DIRNAME, true);
+  if (os_isdir(p))
+    return p;
+  free(p);
+  return NULL;
+}
+
+/// If the string between "p" and "pend" ends in "name/", return "pend" minus
+/// the length of "name/".  Otherwise return "pend".
+static char_u *remove_tail(char_u *p, char_u *pend, char_u *name)
+{
+  int len = (int)STRLEN(name) + 1;
+  char_u      *newend = pend - len;
+
+  if (newend >= p
+      && fnamencmp(newend, name, len - 1) == 0
+      && (newend == p || after_pathsep(p, newend)))
+    return newend;
+  return pend;
+}
+
+/// Vim's version of getenv().
+/// Special handling of $HOME, $VIM and $VIMRUNTIME, allowing the user to
+/// override the vim runtime directory at runtime.  Also does ACP to 'enc'
```

Comment:
ACP is apparently the Ascii Code Page, and it appears to be Windows translation management magic (https://msdn.microsoft.com/en-us/library/windows/desktop/dd318070%28v=vs.85%29.aspx). Rewording this to "`Should do ACP  to 'enc'...`" may not be a terrible idea, since while it is not currently representative of the function's behavior, as with the current directory changing issue in `init_homedir` ACP to `'enc'` conversion seems like a likely pitfall for the Windows porting effort in the future